raiden-ts/src/services/epics.ts

```diff
@@ -695,3 +706,146 @@ export const monitorRequestEpic = (
+
+        assert(balance.sub(amount).gte(Zero), [
+          ErrorCodes.UDC_PLAN_WITHDRAW_EXCEEDS_AVAILABLE,
+          {
+            balance: balance.toString(),
+            amount: amount.toString(),
+          },
+        ]);
+      } catch (e) {
+        return of(udcWithdraw.failure(e, action.meta));
+      }
+      const contract = getContractWithSigner(userDepositContract, signer);
+      return from(contract.functions.planWithdraw(amount)).pipe(
+        assertTx('planWithdraw', ErrorCodes.UDC_PLAN_WITHDRAW_FAILED, { log }),
+        concatMap((txHash) =>
+          combineLatest([
+            from(provider.getTransactionReceipt(txHash)),
+            state$.pipe(pluckDistinct('blockNumber')),
+            config$.pipe(pluckDistinct('confirmationBlocks')),
+          ]).pipe(
+            first(
+              ([receipt, blockNumber, confirmationBlocks]) =>
+                !!receipt.blockNumber && receipt.blockNumber + confirmationBlocks <= blockNumber,
+            ),
+            map(() => txHash),
+          ),
+        ),
+        concatMap((txHash) => {
+          return from(userDepositContract.functions.withdraw_plans(address)).pipe(
+            map(({ amount, withdraw_block }) =>
+              udcWithdrawPlanned(
+                {
+                  amount: amount as UInt<32>,
+                  block: withdraw_block as UInt<32>,
+                },
+                {
+                  txHash,
+                },
+              ),
+            ),
+          );
+        }),
+        catchError((err) => {
+          log.error('Planning udc withdraw failed', err);
+          return of(udcWithdraw.failure(err, action.meta));
+        }),
+      );
+    }),
+  );
+};
+
+export const udcCheckWithdrawPlannedEpic = (
+  action$: Observable<RaidenAction>,
+  state$: Observable<RaidenState>,
+  { userDepositContract, address }: RaidenEpicDeps,
+): Observable<udcWithdrawPlanned> => {
+  return action$.pipe(
+    filter(newBlock.is),
+    first(),
+    concatMap(() =>
+      from(userDepositContract.functions.withdraw_plans(address)).pipe(
+        filter((value) => value.withdraw_block.gt(Zero)),
+        map(({ amount, withdraw_block }) =>
+          udcWithdrawPlanned(
+            {
+              amount: amount as UInt<32>,
+              block: withdraw_block as UInt<32>,
+            },
+            {},
+          ),
+        ),
+      ),
+    ),
+  );
+};
+
+export const udcWithdrawPlannedEpic = (
+  action$: Observable<RaidenAction>,
+  state$: Observable<RaidenState>,
+  { log, userDepositContract, address, signer }: RaidenEpicDeps,
+): Observable<udcWithdraw.success | udcWithdraw.failure> => {
+  return action$.pipe(filter(udcWithdrawPlanned.is)).pipe(
+    mergeMap((action) =>
+      state$.pipe(
+        pluck('blockNumber'),
+        first((blockNumber) => action.payload.block.lte(blockNumber)),
+        map(() => action),
+      ),
+    ),
+    withLatestFrom(from(userDepositContract.functions.balances(address))),
```

Comment:
`withLatestFrom` again..

Reply:
The same as above with mergeMap?:

Reply:
Yes, but also see the `defer` comment below